src/component-library/Flex/Flex.style.tsx

```diff
@@ -18,7 +18,8 @@ const StyledFlex = styled.div<StyledFlexProps>`
   flex-direction: ${(props) => props.$direction};
   justify-content: ${(props) => props.$justifyContent};
   align-items: ${(props) => props.$alignItems};
-  gap: ${(props) => theme.spacing[props.$gap]};
+  // MEMO: for handling spacing0
+  gap: ${(props) => (theme.spacing as Record<Spacing, string>)[props.$gap]};
```

Comment:
Not a blocker, but I'd prefer not to cast this (although I assume there's a reason why you have to!). If it's not possible to resolve now could you add a TODO to refactor a bit further down the line?

Reply:
@tomjeatt I can remove the cast if we add spacing0 to the theme file. Does it make sense having there `spacing0: ""`

Reply:
Yes, much happier with that :) spacing0 is a valid value, makes sense to have it in the theme file